Src/BootCamp.Chapter/IEnumerableExtensions.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace BootCamp.Chapter
+{
+    public static class IEnumerableExtensions
+    {
+        public static IEnumerable<T> SnapFingers<T>(this IEnumerable<T> list, Predicate<T> predicate)
+        {
+            var list2 = list.Where(x => predicate(x)).ToList();
+            var list3 = list2.RemoveHalfElementsAtRandom();
+
+            return list3; 
+        }
+
+        public static List<T> RemoveHalfElementsAtRandom<T>(this IEnumerable<T> list)
+        {
+            var indexOfHalfList = Math.Floor((float)list.Count() / 2);
+            var convertedToList = list.ToList();
```

Comment:
You don't have to convert to list. It's not so obvious, but you can foreach every element, increment index i and break when i reaches half of total.

It's fine for now, I am sure you can do it, so no worries.